src/recordmerge.jl

```diff
@@ -0,0 +1,13 @@
+"""
+    recordmerge(record; fields_to_merge...)
+
+Return an `AbstractRecord` by merging the fields of `record` and `fields_to_merge` together.
+
+See also: [`Tables.rowmerge`](@ref).
+"""
+function recordmerge(record::AbstractRecord; fields_to_merge...)
+    # Avoid using `typeof(record)` as can cause constructor failures with parameterized 
+    # record types.
+    R = Legolas.record_type(Legolas.schema_version_from_record(record))
```

Comment:
```suggestion
    R = record_type(schema_version_from_record(record))
```
No need to qualify with the module name inside of the module since there are no locally conflicting names

Reply:
My bad. I forgot to remove this after upstreaming this change.